### ISSUE TITLE
Lost funds

### ISSUE THREAD

**Issue by chefindex** [LN-Zap/zap-iOS#362]
Good morning. I sent this last week to support through the app but haven't received anything back yet. 

"What’s up Jack? If this is actually you at this point. I wanna say I love the contribution of bitcoin to develop the research of Bitcoin development in Africa. I believe it will be the future of the world because there are less walls to tear down there to build new ones. 

Regarding my problem? First let me ask one question? Do I have to deposit lighting Bitcoin not just Bitcoin.  If that’s the case which I think it is I might have messed up. I tried to send $567.77 to my zap wallet last night from a blue wallet. The transaction as been confirmed on the blue wallet side but no funds on the zap side. I’m not tech savvy to say the least but I would like to use your products to further your development if I can. Let me know if you can help. 

https://blockstream.info/tx/0e9997164848d542a3e516b608c84e98cad9b08d86b015a2b4bb5da1759ec258

![image](https://user-images.githubusercontent.com/78431325/108756372-464e0c00-74fd-11eb-8b2e-0a9889b81edc.jpeg)

Derek Orrell
Beck & Call Catering NYC & LA
Exec. Chef/Owner
(917)405-9624
derek@beckandcallnyc.com
www.beckandcallnyc.com
@beckandcallnyc

**Comment by chill117** [comment]
First, you should remove your contact information from this post.

Couple questions for you:
* Do you see any transactions in the Zap iOS app? There is a button in the top right corner that looks like a clock.
* How did you generate the receiving address in your Zap iOS app?

**Comment by chefindex** [comment]
Thanks for that. I copied an email that I sent to support an didn't know I did that. Man honestly just answer me one question. I think I messed up. Was it supposed to be from a lighting wallet? And if it wasn't would it just go somewhere else?

**Comment by chill117** [comment]
> Was it supposed to be from a lighting wallet? And if it wasn't would it just go somewhere else?

The funds can be sent from any other bitcoin wallet - that is not so important. What is important: Which address did you send the bitcoin to? Which wallet did you use to send the funds?

**Comment by chefindex** [comment]
My funds were sent to this address on zap. Simply pushed the receive button and copied the address. I've used numerous other wallets and haven't had this issue. Please let me know what I can do.

**Comment by chefindex** [comment]
bc1q7yaf7spjr5ne8rusmt4mj475gpnhcdrp4qvev9

**Comment by chefindex** [comment]
It was a blue wallet. Mobile.

**Comment by chill117** [comment]
That address has no transactions:
https://blockstream.info/address/bc1q7yaf7spjr5ne8rusmt4mj475gpnhcdrp4qvev9

Check the transaction history in your Blue Wallet app. Are you sure that you sent funds out?

**Comment by chefindex** [comment]
And the money I already sent?

**Comment by chefindex** [comment]
Every time I press the receive button a different address is produced? Why is it doing that? Do you work for Zap?

**Comment by chill117** [comment]
> Every time I press the receive button a different address is produced? Why is it doing that? Do you work for Zap?

No, I do not work for Zap. But I am a developer and I have been using the Zap wallet app for a couple years.

> And the money I already sent?

Again, did you check the transaction history in your Blue Wallet app? How do you know the address was from your Zap wallet?

**Comment by chefindex** [comment]
I own both wallets!!!! I just was trying to send money into the zap wallet to start using it. So I copied it and pasted it myself. Sending money from my own blue wallet to my own zap wallet. Why would the zap app produce a different address?

**Comment by chefindex** [comment]
But this transaction did go through yeah?  

https://blockstream.info/tx/0e9997164848d542a3e516b608c84e98cad9b08d86b015a2b4bb5da1759ec258

Derek Orrell
(917)405-9624
derek.orrell@gmail.com

**Comment by chill117** [comment]
You should remove your contact details from your last comment.

> But this transaction did go through yeah?

Yes, that transaction is confirmed with funds sent to two separate addresses:
```
bc1qg332270sfzvv3l3fdq0zxhqsnnkxvxq93m55pg
0.01 BTC

bc1q5hqtzsulgh494xg49ax0l8tjm6ecw9g9jfw7jz
0.01523085 BTC
```
Most likely the address that received 0.01 BTC is the intended receiving address and the other one is the change sent back to your Blue Wallet app.

> I own both wallets!!!! I just was trying to send money into the zap wallet to start using it. So I copied it and pasted it myself. Sending money from my own blue wallet to my own zap wallet. Why would the zap app produce a different address?

I understand that you are using both of those wallet apps. But every wallet app is different in subtle ways. Zap wallet uses lnd under-the-hood. Which is why a new address is generated every time that you initiate the on-chain receive process by pressing the "Receive" button.

There are two possible explanations for your problem.

The first possibility is user error. You simply copy/pasted the wrong address. Maybe you had an address already in your clipboard ready to be pasted and you didn't actually copy the address from Zap wallet.


The second possibility is a bit more complicated. Let's say you did copy/paste the address correctly from Zap wallet. But you generated many addresses (without using them) in Zap wallet. When a new address is generated an internal counter is incremented by 1 - let's call this the "address index." The address index starts at 0 for the first address, then 1 for the next, and so on. So when the app scans for received funds, it's incrementing this address index to a set limit - let's call this the "gap limit." It's possible that the address you sent funds to is beyond Zap's gap limit. To recover your funds, you will need the seed that was generated when you first setup your Zap wallet. Then you import the seed to [Electrum](https://electrum.org/#home). Electrum should start to automatically scan addresses for funds. If it doesn't find any, you can manually set the "gap limit" higher via Electrum's preferences menu.

**Comment by chefindex** [comment]
Hey man thanks a million for helping me out. Don't even have a way to thank you but I appreciate the help. I will try what you said.

**Comment by chefindex** [comment]
Honestly could it just be taking forever to sync and find the transaction?

**Comment by chill117** [comment]
> Honestly could it just be taking forever to sync and find the transaction?

Could be that also. But if I was in your situation, I would definitely just try to recover funds by importing the seed to Electrum. It gives a lot more control and visibility into what's going on.

**Comment by chefindex** [comment]
I'm on it. Thanks a ton man. I'll update you when I figure it out. Appreciate the help.

**Comment by chefindex** [comment]
Alright so I downloaded Electrum and started up a wallet but it didn't recognize the seed from the zap account. I then did the watch only zap with the address and nothing came up. I check the preferences to look for the gap limit but can't find it.

**Comment by chill117** [comment]
> Alright so I downloaded Electrum and started up a wallet but it didn't recognize the seed from the zap account. I then did the watch only zap with the address and nothing came up. I check the preferences to look for the gap limit but can't find it.

Hmmm. I remember it being there in the interface. But in any case, you can use the following command in Electrum's console:
```py
wallet.change_gap_limit(30)
```
The default value is 20. So start with 30 and slowly increase it until you find the funds.

You can open Electrum's console at View -> Show Console

**Comment by chefindex** [comment]
The only thing is the seed isn't recognized still. So the zap wallet is only a watching only.

**Comment by chefindex** [comment]
I changed the stop gap but I really don't know where to go from there.

**Comment by GarettBill** [comment]
Same thing happened to me but thanks to Blackhathacker111 who helped me message him on telegram his telegram username is https://t.me/Blackhathacker111

**Comment by DrrDino** [comment]
> Good morning. I sent this last week to support through the app but haven't received anything back yet.
> 
> "What’s up Jack? If this is actually you at this point. I wanna say I love the contribution of bitcoin to develop the research of Bitcoin development in Africa. I believe it will be the future of the world because there are less walls to tear down there to build new ones.
> 
> Regarding my problem? First let me ask one question? Do I have to deposit lighting Bitcoin not just Bitcoin. If that’s the case which I think it is I might have messed up. I tried to send $567.77 to my zap wallet last night from a blue wallet. The transaction as been confirmed on the blue wallet side but no funds on the zap side. I’m not tech savvy to say the least but I would like to use your products to further your development if I can. Let me know if you can help.
> 
> https://blockstream.info/tx/0e9997164848d542a3e516b608c84e98cad9b08d86b015a2b4bb5da1759ec258
> 
> ![image](https://user-images.githubusercontent.com/78431325/108756372-464e0c00-74fd-11eb-8b2e-0a9889b81edc.jpeg)

Same thing happened to me but thanks to Blackhathacker111 who helped me recover all my funds,message him on telegram his telegram username is  https://t.me/Blackhathacker111
